File: backend/scripts/test_db.js
const { Client } = require('pg');
const path = require('path');

const projectRoot = path.resolve(__dirname, '..');
require('dotenv').config({ path: path.join(projectRoot, '.env') });

async function test() {
  if (!process.env.DATABASE_URL) {
    console.error('Missing DATABASE_URL in .env');
    process.exit(1);
  }

  const dbUrl = new URL(process.env.DATABASE_URL);
  console.log('Testing connection to:', `${dbUrl.hostname}:${dbUrl.port || '5432'}`);
  const client = new Client({ 
    connectionString: process.env.DATABASE_URL,
    ssl: { rejectUnauthorized: false }
  });



  try {
    await client.connect();
    console.log('Successfully connected!');
    const res = await client.query('SELECT NOW()');
    console.log('Time from DB:', res.rows[0].now);
  } catch (err) {
    console.error('Connection failed:', err.message);
  } finally {
    await client.end();
  }
}
test();

File: backend/scripts/test_db.go
package main

import (
	"database/sql"
	"fmt"
	"os"
	"path/filepath"

	"github.com/joho/godotenv"
	_ "github.com/lib/pq"
	"nyay-api/infrastructure/postgres"
)

func main() {
	projectRoot, err := findProjectRoot()
	if err != nil {
		fmt.Printf("Could not locate project root: %v\n", err)
		return
	}

	_ = godotenv.Load(filepath.Join(projectRoot, ".env"))

	dbURL := postgres.NormalizeDSN(os.Getenv("DATABASE_URL"))
	if dbURL == "" {
		fmt.Println("Missing DATABASE_URL in .env")
		return
	}

	fmt.Println("Testing Supabase Postgres connection from DATABASE_URL")

	db, err := sql.Open("postgres", dbURL)
	if err != nil {
		fmt.Printf("Open failed: %v\n", err)
		return
	}
	defer db.Close()

	err = db.Ping()
	if err != nil {
		fmt.Printf("Ping failed: %v\n", err)
		return
	}

	fmt.Println("Successfully connected to Supabase Postgres via Go!")
}

func findProjectRoot() (string, error) {
	dir, err := os.Getwd()
	if err != nil {
		return "", err
	}

	for {
		if fileExists(filepath.Join(dir, "go.mod")) && fileExists(filepath.Join(dir, "cmd", "api", "main.go")) {
			return dir, nil
		}

		parent := filepath.Dir(dir)
		if parent == dir {
			return "", fmt.Errorf("backend root not found")
		}
		dir = parent
	}
}

func fileExists(path string) bool {
	info, err := os.Stat(path)
	return err == nil && !info.IsDir()
}
